rfcs/0140-simple-package-paths.md

```diff
@@ -0,0 +1,234 @@
+---
+feature: simple-package-paths
+start-date: 2022-09-02
+author: Silvan Mosberger
+co-authors: Nixpkgs Architecture Team
```

Comment:
A team is not an author, because an author is an individual.
Not just semantics, because we should have a good record of who actually participated in the drafting of this RFC.
Might have to dig through the notes to make this complete...

```suggestion
co-authors: Thomas Bereknyei, Robert Hensing, John Ericson, Alex Ameen
```

Reply:
Not entirely related to this review, but we discussed how to unblock this RFC in the NAT meeting. Since only me and @roberth contributed text to this RFC, only we are authors, but we can still list the other members of the team as reviewers like this:

```suggestion
co-authors: Robert Hensing
pre-RFC reviewers: Thomas Bereknyei, John Ericson, Alex Ameen
```

This way the reviewers (@aakropotkin said he can do it) can nominate themselves as shepherds without directly violating the RFC rules. I'm not a big fan of doing this, because all the reviewers from the Nixpkgs Architecture Team are obviously already in favor of the RFC. But this RFC has been open for 2 months now and we still don't have enough shepherds (I also reached out [on Discourse](https://discourse.nixos.org/t/shepherds-needed-for-rfc-140/26543)). I don't think this RFC should be blocked by this any longer, especially since a team of 5 people already discussed it for months and reached consensus about it.

Reply:
> I'm not a big fan of doing this, because all the reviewers from the Nixpkgs Architecture Team are obviously already in favor of the RFC.

I think self-nominating, disclosing «currently siding towards FCP/accept after a week or two», and pledging to give serious consideration to any novel arguments for modification/rejection is perfectly fine.

We have had RFCs with long pre-nomination discussions where people self-nominated with a stated position. If you sincerely say that the effective contributions of the remaining team members are comparable to pre-nomination reviews in other RFCs, it's all fine; you did pre-submission round of reviews in a non-standard way, but you should not be burdened too much more than if you had submitted as a draft and then only three people showed up and then you marked it ready for review after a month of polishing. 

I have made a call for «opposing» shepherds, if we get none, maybe it is a sign of something?

Reply:
I don't think there is a meaningful difference between a co-author and someone who was present at the authoring sessions, but did not physically type any of the text. Someone who is part of the NAT will likely not be able to bring a fresh perspective as opposed to someone entirely new.

Reply:
It feels overly rigorous to me to not be able to progress on a universally-liked idea with 7 reviewers (5 NAT members and 2 shepherds), just because we can't find another person. I'd say we should leave it up to the @NixOS/rfc-steering-committee to decide on this.

Reply:
Our RFC process has the problem that its model of how the process looks like simply does not match reality anymore. I've followed or participated in a couple of RFCs so far, and in many times the proposal was largely reworked or completely rewritten by the shepherd team.

The fact that people organize *before* writing the RFC means that it is likely to be more complete, thought-through and higher quality overall. I see this as a win, and our formalized RFC process actively discourages doing so which is unfortunate.

> Also we measure interest (or lack thereof) by getting together a shepherd team.

I'm sorry, but "interest" is a horrible metric for an RFC. There is a lot of boring, thankless, tedious work to that few people are interested in.

----

I honestly don't know what to say about this. I've been deeply dissatisfied with out current process so far on many accounts, and we should have a larger discussion about this elsewhere. It just would be unfortunate to block RFC 140 on that discussion though.

Reply:
> I'm sorry, but "interest" is a horrible metric for an RFC. There is a lot of boring, thankless, tedious work to that few people are interested in.

https://github.com/NixOS/rfcs/pull/130, what can I say.

Regarding the main point though, to play devil's advocate: Say you have 5 people in the NAT, say you have an RFC that has an author from the NAT (nominally or actually) — then you also have 4 shepherds and a conveniently regularly scheduled meeting of the shepherds — you can go straight to FCP basically.

Reply:
Honestly? I don't have a problem with that. IME people don't care and give feedback until FCP anyways …

Reply:
Thanks a lot for [your nomination](https://github.com/NixOS/rfcs/pull/140#issuecomment-1512080692) as a shepherd @piegamesde! This avoids this problem for now, but we should really have a more proper discussion about this in the future. I'll consider creating an RFC to improve the RFC process around this.